src/features/soft_navigations/aggregate/index.js

```diff
@@ -45,8 +46,13 @@ export class Aggregate extends AggregateBase {
 
     this.blocked = false
     this.waitForFlags(['spa']).then(([spaOn]) => {
-      if (spaOn) this.scheduler.startTimer(harvestTimeSeconds, 0)
-      else this.blocked = true // if rum response determines that customer lacks entitlements for spa endpoint, this feature shouldn't harvest
+      if (spaOn) {
+        this.drain()
+        this.scheduler.startTimer(harvestTimeSeconds, 0)
```

Comment:
This feature is a little different too. It also exposes the harvest scheduler but it creates the harvest scheduler instance outside and before the `waitForFlags`.